.scripts/macos/test.sh

```diff
@@ -2,10 +2,12 @@
 set -e
 set -x
 
+# cd to sources dir
 cd $1
-mkdir -p build
-cd build
 
-# NUM_PROCESSORS=$(cat nprocs.txt)
-make help
-make test_with_output -j ${NUM_PROCESSORS}
\ No newline at end of file
+cd test
+NUM_PROCESSORS=$(cat ../build/nprocs.txt)
+python3 run_tests.py -f -j ${NUM_PROCESSORS}
+
+cd ../build
+make test_with_output
```

Comment:
why did you drop the -j here

Reply:
`NUM_PROCESSORS` was undefined before resulting in `-j` with no max jobs, which you never really want. Besides it is redundant because the tests are all built at this point and the test running component which this is driving can't be parallelized.